pkg/compare/path.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package compare
+
+import "strings"
+
+// Path provides a JSONPath-like struct and field-member "route" to a
+// particular field within a compared struct
+type Path struct {
+	parts []string
+}
+
+// Push adds a new part to the Path.
+func (p Path) Push(part string) {
+	p.parts = append(p.parts, part)
+}
+
+// Pop removes the last part from the Path
+func (p Path) Pop() {
+	p.parts = p.parts[:len(p.parts)-1]
```

Comment:
Should there be an empty check? is this prone to OutOfBoundsException currently?

Reply:
Nice catch. Adding test...